subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperChecksumVerificationTest.groovy

```diff
@@ -72,4 +75,53 @@ Expected checksum: 'bad'
         then:
         success.output.contains('BUILD SUCCESSFUL')
     }
+
+    def "wrapper update require checksum if it was provided in properties"() {
+        given: "Init wrapper with under development distribution"
+        prepareWrapper(new URI("${server.uri}/gradle-bin.zip"))
+
+        and: "checksum is provided in properties"
```

Comment:
Documenting Spock `setup/when/then` tags are not usually used in the Gradle codebase. I'm not against it, but be careful not to over-document things.

```suggestion
        and: "Checksum is provided in properties"
```